sim/shaman/elemental/elemental.go

```diff
@@ -83,9 +83,10 @@ func (eleShaman *ElementalShaman) Act(sim *core.Simulation) time.Duration {
 		// Only way for a shaman spell to fail is due to mana cost.
 		// Wait until we have enough mana to cast.
 		regenTime := eleShaman.TimeUntilManaRegen(newAction.GetManaCost())
-		newAction = core.NewWaitAction(sim, eleShaman.GetCharacter(), regenTime)
+		newAction = core.NewWaitAction(sim, eleShaman.GetCharacter(), regenTime, core.WaitReasonOOM)
+		newAction.Cast(sim)
```

Comment:
Somewhere this cast was already being called, but not every time. I am not sure what path results in it being cast but we should make sure we aren't double casting

Reply:
AFAICT cast wasn't called anywhere. Until this point it didn't matter of course because WaitActions didn't do anything.